### DISCUSSION TITLE
Record access value by name

### DISCUSSION THREAD

**Discussion by cd155** [JacquesCarette/Drasil#3291] [Show and tell]
We can access the value in a record by a name. Is the record fundamental the same as the dictionary/hash table? They all access values by name.

**Comment by JacquesCarette** [comment]
Records can be statically checked (at compile time) to have the right fields. Dictionaries/hash table are dynamic structures that must do this checking at run-time, and this check can fail.

They are only similar at a quite shallow level.

**Reply by cd155** [reply]
Thanks for explaining it in detail.

This promotes me think about why functions are accessed by both `name` and `distance`. I feel almost all programming languages, we call the function by its name and give arguments. Those arguments will be accessed by distance, like the first argument, the second... I just wonder why the function will be used in this way. Does it become a common practice?

**Reply by JacquesCarette** [reply]
Because we (humans) can do 1-2-3 very easily, but 6-7-8 is already hard, 11-12-13 is awful. (I mean those numbers as positions.) Just because computers can do positional trivially doesn't me we can.  Furthermore, having 7 arguments (say) that all are `int` but all mean something **different** is a recipe for subtle bugs, as it is too trivial to permute things.

We, humans, work much better with signals of intent, which are easiest done via words, in this case, field names in records.

**Reply by cd155** [reply]
Should the function don't use positional access at all? Arguments could access by name, right?

**Reply by JacquesCarette** [reply]
By-name-only would be awkward for 1-argument functions. By position is ok for 1-3 arguments of different types.
